plugins/thumbnail/lib/model/thumbnailFactory/kImageTransformationAdapter.php

```diff
@@ -17,18 +19,96 @@ class kImageTransformationAdapter
+		{
+			return $this->getStripTransformation();
+		}
+
 		$step = new kImageTransformationStep();
-		$this->createEntrySource($step);
+		$this->addSource($step);
 		$this->handleSourceActions($step);
-		$this->prepareInput();
+		$this->handleActionType($step);
+		$this->handleStripProfile($step);
+		$this->handleImageOutputAction($step);
+		$transformation = new kImageTransformation();
+		$transformation->addImageTransformationStep($step);
+		return $transformation;
+	}
+
+	protected function getStripTransformation()
+	{
+		$transformation = new kImageTransformation();
+		$vidSlices = $this->parameters->get(kThumbFactoryFieldName::VID_SLICES);
+		$startSec = $this->parameters->get(kThumbFactoryFieldName::START_SEC);
+		$interval = $this->getStripInterval();
+		$startSec = max($startSec, 0);
+		$step = $this->getFirstStripTransofmrationStep($startSec, $vidSlices);
+		$transformation->addImageTransformationStep($step);
+		for($i = 1; $i <= $vidSlices; $i++)
+		{
+			$step = new kImageTransformationStep();
+			$this->addEntrySource($step);
+			$this->addVidSecAction($step, $startSec + ($i * $interval));
+			$this->handleActionType($step);
+			$this->addStripConcat($step, $i);
+			$transformation->addImageTransformationStep($step);
+		}
+
+		$this->handleStripProfile($step);
+		$this->handleImageOutputAction($step);
+		return $transformation;
+	}
+
+	protected function getStripInterval()
+	{
+		$entry = $this->parameters->get(kThumbFactoryFieldName::ENTRY);
+		$videoLength = $entry->getLengthInMsecs();
+		$startSec = $this->parameters->get(kThumbFactoryFieldName::START_SEC);
+		$endSec =  $this->parameters->get(kThumbFactoryFieldName::END_SEC);
+		$vidSlices = $this->parameters->get(kThumbFactoryFieldName::VID_SLICES);
+		if($startSec !==  kThumbAdapterParameters::UNSET_PARAMETER && $endSec !==  kThumbAdapterParameters::UNSET_PARAMETER)
+		{
+			$interVal = $this->endSec - $this->startSec / $vidSlices;
+		}
+		else
+		{
+			$interVal = $videoLength / self::MILLISECONDS_IN_SECOND / $vidSlices;
+		}
+
+		return $interVal;
+	}
+
+	/**
+	 * @param kImageTransformationStep $step
+	 * @param int $slideNum
+	 */
+	protected function addStripConcat($step, $slideNum)
+	{
+		$concatAction = new kCompositeAction();
+		$concatAction->setActionParameter(kThumbnailParameterName::X, $slideNum * $this->parameters->get(kThumbFactoryFieldName::WIDTH));
+		$step->addAction($concatAction);
+	}
+
+	protected function handleActionType($step)
+	{
 		switch($this->parameters->get(kThumbFactoryFieldName::TYPE))
 		{
 			case kExtwidgetThumbnailActionType::RESIZE:
+				if($this->parameters->get(kThumbFactoryFieldName::WIDTH)  > kResizeAction::BEST_FIT_MIN && $this->parameters->get(kThumbFactoryFieldName::HEIGHT) > kResizeAction::BEST_FIT_MIN)
+				{
+					$this->handleResize(true, $step);
+				}
+				else
+				{
+					$this->handleResize(false, $step);
+				}
+				break;
```

Comment:
Just a suggest for shorter code- 
$bestFit = $this->parameters->get(kThumbFactoryFieldName::WIDTH)  > kResizeAction::BEST_FIT_MIN && $this->parameters->get(kThumbFactoryFieldName::HEIGHT) > kResizeAction::BEST_FIT_MIN;
$this->handleResize($bestFit , $step);